Content.Server/Stunnable/Systems/StunbatonSystem.cs

```diff
@@ -100,17 +108,23 @@ private void TurnOff(EntityUid uid, StunbatonComponent comp)
 
         private void TurnOn(EntityUid uid, StunbatonComponent comp, EntityUid user)
         {
+
             if (comp.Activated)
                 return;
 
             var playerFilter = Filter.Pvs(comp.Owner, entityManager: EntityManager);
             if (!TryComp<BatteryComponent>(comp.Owner, out var battery) || battery.CurrentCharge < comp.EnergyPerUse)
             {
+
                 SoundSystem.Play(comp.TurnOnFailSound.GetSound(), playerFilter, comp.Owner, AudioHelpers.WithVariation(0.25f));
                 user.PopupMessage(Loc.GetString("stunbaton-component-low-charge"));
                 return;
             }
 
+            if (TryComp<RiggedComponent>(uid, out var rig) && rig.IsRigged)
+                    _riggedSystem.Explode(uid, battery, user);
+
```

Comment:
Indented by 4 more than it needs to be.

Also this is going to kill it so no point continuing.